### PR TITLE
Upgrade GitHub Actions

### PR DESCRIPTION
**Pull request by cclauss**
https://github.com/actions/checkout/releases
https://github.com/actions/setup-python/releases
https://github.com/pypa/gh-action-pypi-publish/releases

### CODE REVIEW

**Comment by wch** [comment]
I made some changes to the workflow on `main`, because these repositories are now public. Can you rebase onto `main` and push?

File: .github/workflows/pytest.yaml [hunk 1]
```diff
@@ -98,7 +98,7 @@ jobs:
 
       # test deploy ----
       - name: "Test Deploy to PyPI"
-        uses: pypa/gh-action-pypi-publish@release/v1
+        uses: pypa/gh-action-pypi-publish@v1
```
Comment (wch) [inline .github/workflows/pytest.yaml:101]: Are you sure this is right? I see a branch named `release/v1`, but not one named `v1`. (And same for the other instance below.)

Comment (cclauss) [reply .github/workflows/pytest.yaml:101]:
```suggestion
        uses: pypa/gh-action-pypi-publish@release/v1
```

File: .github/workflows/pytest.yaml [hunk 2]
```diff
@@ -107,7 +107,7 @@ jobs:
 
       ## prod deploy ----
       - name: "Deploy to PyPI"
-        uses: pypa/gh-action-pypi-publish@master
+        uses: pypa/gh-action-pypi-publish@v1
```
Comment (cclauss) [inline .github/workflows/pytest.yaml:110]:
```suggestion
        uses: pypa/gh-action-pypi-publish@release/v1
```

**Comment by wch** [comment]
Thanks!

**Pull request merged**